accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,188 @@
+            </li>
+          </div>
+        </ol>
+      </li>
+      <toolbox-tree-view *ngIf= "block.nextConnection && block.nextConnection.targetBlock()" [level]="level" [block]="block.nextConnection.targetBlock()" [displayBlockMenu]="false"></toolbox-tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.ToolboxTreeView; }),
+        blocklyApp.FieldView],
+    inputs: ['block', 'displayBlockMenu', 'level', 'index', 'tree'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService,
+                  function(_clipboardService, _treeService) {
+      this.infoBlocks = {};
+      this.nextBlock = {};
+      this.sharedClipboardService = _clipboardService;
+      this.treeService = _treeService;
+    }],
+    setLabelledBy: function(item, id) {
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', id);
+      }
+    },
+    concatStringWithSpaces: function() {
+      var string = arguments[0];
+      for (i = 1; i < arguments.length; i++) {
+        var arg = arguments[i];
+        // arg can be undefined and if so, should be ignored.
+        if (arg) {
+          string += ' ' + arg;
+        }
+      }
+      return string;
+    },
+    createCategoryDependantId: function(index, parentList) {
+      // If this is the first block in a category-less toolbox,
+      // the id should be blockly-toolbox-tree-node0.
+      if (index === 0) {
+        if (this.tree.getAttribute('aria-activedescendant') ==
+            'blockly-toolbox-tree-node0') {
+          this.addClass(parentList, 'blocklyActiveDescendant');
+          parentList.setAttribute('aria-selected', 'true');
+          this.treeService.setActiveDesc(parentList, this.tree.id);
+        }
+        return 'blockly-toolbox-tree-node0';
+      } else {
+        return this.treeService.createId(parentList);
+      }
+    },
+    addClass: function(node, classText) {
+      // Ensure that node doesn't have class already in it.
+      var classList = node.className;
+      classList = classList.split(' ');
+      var canAdd = true;
+      for (var i = 0; i < classList.length; i++) {
+        if (classList[i] == classText) {
+          canAdd = false;
+        }
+      }
+      // Add class if it doesn't.
+      if (canAdd) {
+        if (classList.length) {
+          node.className += classText;
+        } else {
+          node.className += ' ' + classText;
+        }
+      }
+    },
+    getInfo: function(block) {
+      // Get the list all inputs.
+      if (this.infoBlocks[block.id]) {
+        this.infoBlocks[block.id].length = 0;
+      } else {
+        this.infoBlocks[block.id] = [];
+      }
+
+      var blockInfoList = this.infoBlocks[block.id];
+
+      for (var j = 0, field; field = block.fieldRow[j]; j++) {
+        blockInfoList.push(field);
+      }
+
+      return this.infoBlocks[block.id];
```

Comment:
I don't see the point of using infoBlocks. If I'm reading this correctly, shouldn't this function read more like:

    return block.fieldRow;

?